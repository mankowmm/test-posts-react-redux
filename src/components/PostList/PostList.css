.post-list {
    margin: 30px;
}

.post-list-header, .post-list-row {
    display: flex;
}

.post-list-header {
    border: 1px solid gray;
    font-weight: bold;
}

.post-list-header > div, .post-list-row > div {
    width: 33.3%;
    padding: 10px;
}

.post-list-row > div {
    border:  1px dashed #c0c0c0;
}