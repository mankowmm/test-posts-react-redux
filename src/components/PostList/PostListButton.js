import React, { Component } from 'react';
import { Link } from 'react-router-dom'

class PostLink extends Component {
    render() {
        return <Link className="btn btn-secondary" to={{pathname: `/post/${this.props.post.id}`}}>See detail</Link>
    }
}

export default PostLink;
