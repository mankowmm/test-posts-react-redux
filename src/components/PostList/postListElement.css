.PostListElement {
    width: calc(100% - 30px);
    margin: 15px;

}

@media screen and (min-width: 600px) {
    .PostListElement {
        width: calc(50% - 30px);
    }
}

@media screen and (min-width: 1024px) {
    .PostListElement {
        width: calc(33.3% - 30px);
    }
}

