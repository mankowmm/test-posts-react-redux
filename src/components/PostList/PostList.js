import React from 'react';
import './PostList.css';
import { Loader } from '../Loader/Loader'
import { PostListSearch } from '../PostListSearch/PostListSearch'

export const PostList = ({posts, loading}) => {
    
    const renderPostListRows = posts.map((post) => {
        return (
            <div key={post.id} className="post-list-row">
                <div>{post.id}</div>
                <div>{post.title}</div>
                <div>{post.body}</div>
            </div>
        )
    })
    
    const retDom = loading ? (
        <Loader/>
      ) : (
        <div className="post-list">
            <PostListSearch/>
            <div className="post-list-header">
                <div>ID</div>
                <div>Title</div>
                <div>Body</div>
            </div>
            {renderPostListRows}
        </div>
      );

    return (
        retDom
    )
}