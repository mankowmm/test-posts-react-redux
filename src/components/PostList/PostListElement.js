import React, { Component } from 'react';
import { PostListButton } from './PostListButton';
import { Card, CardBody, CardTitle, CardSubtitle } from 'reactstrap';
import './postListElement.css';

export class PostListElement extends Component {
    render() {
        return (
            <div className="PostListElement">
                <Card>
                    <CardBody>
                        <CardTitle>
                            <span><b>UserId</b>: </span>
                            <span className="userIdTxt">{this.props.post.userId}</span>
                        </CardTitle>
                        <br/>
                        <CardSubtitle>
                            <span><b>Title</b>: </span>
                            <span className="titleTxt">{this.props.post.title}</span>
                        </CardSubtitle>
                        <br/>
                        <PostListButton post={this.props.post}/>
                    </CardBody>
                </Card>
            </div>
        );
    }
}

