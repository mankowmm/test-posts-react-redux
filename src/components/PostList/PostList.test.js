import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { shallow } from 'enzyme';
import { PostList } from './PostList';

describe('PostList component test suite', ()=> {

    let elWrapper;
    let postsMock;
    let loading;
    let error;

    beforeEach(() => {
        postsMock = [];
        loading = false;
        error = null;
    })
    
    it('should validate with snapshot', () => {
        const tree = renderer
          .create(<PostList loading={loading} posts={postsMock}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render PostList with Loader Component when loading is true passed', ()=> {
        postsMock = [];
        loading = true;
        elWrapper = shallow(<PostList loading={loading} posts={postsMock}/>);
        expect(elWrapper.find('Loader').length).toBe(1)
    })

    it('should render PostList with ErrorMessage component when error passed', ()=> {
        postsMock = [];
        loading = false;
        error = new Error('Network Error');
        elWrapper = shallow(<PostList loading={loading} posts={postsMock} error={error}/>);
        expect(elWrapper.find('ErrorMessage').length).toBe(1)
    })

    it('should render PostList with PostListElement components when passed', ()=> {
        postsMock = [
            {id: 1, title: 'TEST1'},
            {id: 2, title: 'TEST2'},
            {id: 3, title: 'TEST3'}
        ];
        loading = false;
        elWrapper = shallow(<PostList loading={loading} posts={postsMock}/>);
        expect(elWrapper.find('.PostList').length).toBe(1);
        expect(elWrapper.find('PostListElement').length).toBe(3)
    })
    

       

});