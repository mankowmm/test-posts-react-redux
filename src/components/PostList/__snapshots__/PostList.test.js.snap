// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component test suite should validate with snapshot 1`] = `
<div
  className="PostList"
>
  <div
    className="PostListItems"
  />
</div>
`;
